app/controllers/collections_controller.rb

```diff
@@ -95,14 +97,15 @@ def view_shared
   end
 
   def save_shared
-    return redirect_to user_messages_path(current_user), alert: t('.errors.already_member') if @collection.users.include? current_user
+    user = current_user
+    return redirect_to user_messages_path(user), alert: t('.errors.already_member') if @collection.users.include? user
 
-    @collection.users << current_user
-    if @collection.save
-      redirect_to @collection, notice: t('.success_notice')
-    else
-      redirect_to users_messages_path, alert: t('common.errors.something_went_wrong')
-    end
+    @collection.users << user
+    message = Message.received_by(user).find_by(param_type: 'collection', param_id: @collection.id)
+    message.mark_as_deleted(user)
+    message.save
+
+    redirect_to @collection, notice: t('.success_notice')
```

Comment:
So the success is defined now? Saving the message can still fail, no?

Reply:
Added a transaction to ensure database-consistency